File: iOSDemo/iOSDemo/AdDemo/Ad(广告样式)/Native(原生广告)/Resource/MenuVC/NativeTypeSelectVC.m
//
//  NativeTypeSelectVC.m
//  iOSDemo
//
//  Created by ltz on 2025/1/11.
//

#import "NativeTypeSelectVC.h"
#import "SelfRenderVC.h"
#import "ExpressVC.h"
#import "FeedSelfRenderVC.h"
#import "FeedTemplateVC.h"

@interface NativeTypeSelectVC () <UITableViewDelegate,UITableViewDataSource>

@end

@implementation NativeTypeSelectVC

#pragma mark - UITableViewDelegate & UITableViewDataSource
- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return self.dataSource.count;
}

- (CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPath *)indexPath {
    return 120.0; // 每个cell的高度
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    CustomTableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"CustomTableViewCellIdentifier" forIndexPath:indexPath];
    
    if (!cell) {
        cell = [[CustomTableViewCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:@"CustomTableViewCellIdentifier"];
    }
    
    NSDictionary *data = self.dataSource[indexPath.row];
    // 配置cell的titleLbl和subTitleLbl的文本
    cell.titleLbl.text = data[@"title"];
    cell.subTitleLbl.text = data[@"subtitle"];

    return cell;
}
 
- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
    [tableView deselectRowAtIndexPath:indexPath animated:YES];
    NSString *selectedItem = self.dataSource[indexPath.row][@"title"];
    ATDemoLog(@"Selected: %@", selectedItem); 
    if ([selectedItem isEqualToString:kLocalizeStr(@"自渲染广告")]) {
        SelfRenderVC * vc = [SelfRenderVC new];
        vc.title = selectedItem;
        [self.navigationController pushViewController:vc animated:YES];
    }
    if ([selectedItem isEqualToString:kLocalizeStr(@"模板广告")]) {
        ExpressVC * vc = [ExpressVC new];
        vc.title = selectedItem;
        [self.navigationController pushViewController:vc animated:YES];
    }
    if ([selectedItem isEqualToString:kLocalizeStr(@"信息流广告(自渲染)")]) {
        FeedSelfRenderVC * vc = [FeedSelfRenderVC new];
        vc.title = selectedItem;
        [self.navigationController pushViewController:vc animated:YES];
    }
    if ([selectedItem isEqualToString:kLocalizeStr(@"信息流广告(模板渲染)")]) {
        FeedTemplateVC * vc = [FeedTemplateVC new];
        vc.title = selectedItem;
        [self.navigationController pushViewController:vc animated:YES];
    } 
}
  
- (void)viewDidLoad {
    [super viewDidLoad];
    
    [self addNormalBarWithTitle:self.title];
     
    // 初始化数据源
    self.dataSource = @[
        @{@"title": kLocalizeStr(@"自渲染广告"), @"subtitle": kLocalizeStr(@"演示基础的单个自渲染广告")},
        @{@"title": kLocalizeStr(@"模板广告"), @"subtitle": kLocalizeStr(@"演示基础的单个模板广告")},
        @{@"title": kLocalizeStr(@"信息流广告(自渲染)"), @"subtitle": kLocalizeStr(@"在滚动列表单元中展示自定义的广告")},
        @{@"title": kLocalizeStr(@"信息流广告(模板渲染)"), @"subtitle": kLocalizeStr(@"在滚动列表单元中展示模版广告")}
    ];
    
    self.tableView.dataSource = self;
    self.tableView.delegate = self;
    [self.view addSubview:self.tableView];
    [self.tableView mas_makeConstraints:^(MASConstraintMaker *make) {
        make.left.right.bottom.equalTo(self.view);
        make.top.mas_equalTo(self.nbar.mas_bottom);
    }];
}

@end

File: SwiftDemo/SwiftDemo/AdDemo/Native/NativeResources/Other/NativeTypeSelectVC.swift
//
//  NativeTypeSelectVC.swift
//  SwiftDemo
//
//  Created by ltz on 2025/1/11.
//

import UIKit
import SnapKit

class NativeTypeSelectVC: BaseVC {
     
    // MARK: - Lifecycle
    override func viewDidLoad() {
        super.viewDidLoad()
        
        addNormalBar(withTitle: title)
        
        // 初始化数据源
        dataSource = [
            ["title": kLocalizeStr("Self Render Ad"), "subtitle": kLocalizeStr("Demonstrate basic single self-render ad")],
            ["title": kLocalizeStr("Express Ad"), "subtitle": kLocalizeStr("Demonstrate basic single template ad")],
            ["title": kLocalizeStr("Feed Ads (Self Render)"), "subtitle": kLocalizeStr("Display custom ads in scrollable list cells")],
            ["title": kLocalizeStr("Feed Ads (Template Render)"), "subtitle": kLocalizeStr("Display template-style advertisements in scrolling list units.")],
        ]
        
        tableView.dataSource = self
        tableView.delegate = self
        
        view.addSubview(tableView)
        
        tableView.snp.makeConstraints { make in
            make.left.right.bottom.equalTo(view)
            make.top.equalTo(nbar!.snp.bottom)
        }
    }
}

// MARK: - UITableViewDelegate & UITableViewDataSource
extension NativeTypeSelectVC: UITableViewDelegate, UITableViewDataSource {
    
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return dataSource.count
    }
    
    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
        return 120.0 // 每个cell的高度
    }
    
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: "CustomTableViewCellIdentifier", for: indexPath) as! CustomTableViewCell
        
        let data = dataSource[indexPath.row]
        // 配置cell的titleLbl和subTitleLbl的文本
        cell.titleLbl.text = data["title"]
        cell.subTitleLbl.text = data["subtitle"]
        
        return cell
    }
    
    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        tableView.deselectRow(at: indexPath, animated: true)
        
        guard let selectedItem = dataSource[indexPath.row]["title"] else { return }
        ATDemoLog("Selected: \(selectedItem)")
        
        if selectedItem == kLocalizeStr("Self Render Ad") {
            let vc = NativeSelfRenderVC()
            vc.title = selectedItem
            navigationController?.pushViewController(vc, animated: true)
        }
        else if selectedItem == kLocalizeStr("Express Ad") {
            let vc = NativeExpressVC()
            vc.title = selectedItem
            navigationController?.pushViewController(vc, animated: true)
        }
        else if selectedItem == kLocalizeStr("Feed Ads (Self Render)") {
            let vc = FeedSelfRenderVC()
            vc.title = selectedItem
            navigationController?.pushViewController(vc, animated: true)
        }
        else if selectedItem == kLocalizeStr("Feed Ads (Template Render)") {
            let vc = FeedTemplateVC()
            vc.title = selectedItem
            navigationController?.pushViewController(vc, animated: true)
        }
    }
}
